From ef762a7d992b1a0e0b0cb8f0f5961e8f55efad2d Mon Sep 17 00:00:00 2001
From: Matthew Rudy Jacobs <matthewrudyjacobs@gmail.com>
Date: Fri, 23 Apr 2010 22:19:57 +0800
Subject: [PATCH] If assert_recognize is run outside of a functional test
 make sure we get a nice error
 prompting us to do

  setup do
    @routes = Rails.application.routes
  end
---
 .../action_dispatch/testing/assertions/routing.rb  |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/actionpack/lib/action_dispatch/testing/assertions/routing.rb b/actionpack/lib/action_dispatch/testing/assertions/routing.rb
index 1499c03..afdf84f 100644
--- a/actionpack/lib/action_dispatch/testing/assertions/routing.rb
+++ b/actionpack/lib/action_dispatch/testing/assertions/routing.rb
@@ -186,6 +186,9 @@ module ActionDispatch
           request.env["REQUEST_METHOD"] = request_method.to_s.upcase if request_method
           request.path = path
 
+          if @routes.nil?
+            raise "@routes is not set. Tory setting @routes = Rails.application.routes"
+          end
           params = @routes.recognize_path(path, { :method => request.method })
           request.path_parameters = params.with_indifferent_access
 
-- 
1.6.6

